Core/EventHandlers.lua

```diff
@@ -582,6 +582,22 @@ function R:OnIslandCompleted(event, mapID, winner)
 	end
 end
 
+local Scalebane_Keep_QuestID = 70867
+function R:OnQuestTurnedIn(_, questID)
```

Comment:
It seems there are now two functions `OnQuestTurnedIn`, with the second one overwriting the first.

The first one is used to detect the BFA world boss that drops [Slightly Damp Pile of Fur](https://www.wowhead.com/item=174842/slightly-damp-pile-of-fur). If you want to use this event (instead of `SCENARIO_COMPLETED`), you will have to modify the existing event handler:

1. Add the `SCALEBANE_KEEP_SCENARIO_COMPLETION` ID to the existing list of relevant quests
2. Probably rename it, since `worldBossQuests` would no longer be an accurate description

Other than that, it should "just work" since the existing handler would select the toy if it's in the list.

Reply:
ok thats a bit confusing but i think i understand, i would have to add a different function name in the register event section under "QUEST_TURNED_IN" to not overwrite the existing one?

> 1. Add the `SCALEBANE_KEEP_SCENARIO_COMPLETION` ID to the existing list of relevant quests

But also add something similar to to world boss quests? Don't think i quite understand that part

Reply:
but from what i do understand i would do something like 
`local eventQuests = {
	[70867] = "Everlasting Horn of Lavaswimming",
}`

Reply:
> ok thats a bit confusing but i think i understand, i would have to add a different function name in the register event section under "QUEST_TURNED_IN" to not overwrite the existing one?
> 
> > 1. Add the `SCALEBANE_KEEP_SCENARIO_COMPLETION` ID to the existing list of relevant quests
> 
> But also add something similar to to world boss quests? Don't think i quite understand that part

If you re-use the existing event handler, no additional function is needed. Once the toy is added to the list, the event handler should automatically add an attempt for it once the quest was completed. This approach requires minimal changes.

Reply:
ohh ok gotcha, simple then

Reply:
also what would be a suitable rename for "worldBossQuests". Something like "eventQuests" perhaps?

Reply:
> also what would be a suitable rename for "worldBossQuests". Something like "eventQuests" perhaps?

Uhm... I guess that's fine, but maybe `worldEventQuests` instead of `worldBossQuests`?

Reply:
please tell me im not forgetting anything in that latest commit haha

Reply:
Looks good to me. There isn't much to change with this approach. Assuming the event always fires on scenario completion, this version's even simpler than my original idea which required a new event handler. I consider that a good thing :)